models/articles_models.js

```diff
@@ -12,9 +12,19 @@ exports.fetchArticleByID = (id) => {
     })
 }
 
-exports.fetchArticles = () => {
+exports.fetchArticles = (topic) => {
+    const queryValues = [];
+    let queryString = 'SELECT article_id, title, topic, author, created_at, votes, article_img_url FROM articles'
+
+    if(topic) {
+        queryValues.push(topic)
+        queryString += ' WHERE topic = $1'
+    }
+
+    queryString += ' ORDER BY created_at DESC'
+
     return db
-    .query("SELECT article_id, title, topic, author, created_at, votes, article_img_url FROM articles ORDER BY created_at DESC")
+    .query(queryString, queryValues)
     .then(({ rows }) => {
         if(rows.length === 0){
             return Promise.reject({status:404,msg:"No articles found"})
```

Comment:
Looks good. Again, please switch out the length check for an SQL check as a topic may exist but have no articles and we'll still want a 200 in this case. Thanks 🙂